### ISSUE TITLE
TypeError: Value frozenset() with type <class 'frozenset'> is not a valid JAX type

### ISSUE THREAD

**Issue by quattro** [google/lineax#34]
Hi 👋 ,

I'm wondering if it makes sense to annotate the `tags` field in the `MatrixLinearOperator` as,

```python
class MatrixLinearOperator(AbstractLinearOperator):
    """Wraps a 2-dimensional JAX array into a linear operator.

    If the matrix has shape `(a, b)` then matrix-vector multiplication (`self.mv`) is
    defined in the usual way: as performing a matrix-vector that accepts a vector of
    shape `(a,)` and returns a vector of shape `(b,)`.
    """

    matrix: Float[Array, "a b"]
    tags: frozenset[object] = eqx.field(static=True)
    ...
```

This would help avoid the `TypeError` in the title when instances need to be created underneath a `jit`ed function (e.g., inside lax loops).

**Comment by patrick-kidger** [comment]
That seems reasonable! Likewise for many of the other linear operators.

I suppose this hasn't come up too frequently since we always tend to use `eqx.filter_jit` over `jax.jit`.

We'd be happy to take a PR on this!

**Comment by patrick-kidger** [comment]
Closed in #41.
